applications/FluidDynamicsApplication/custom_utilities/mass_conservation_utility.cpp

```diff
@@ -494,18 +600,22 @@ double MassConservationCheckProcess::ComputeFlowOverBoundary( const Kratos::Flag
                         double const w_gauss = gauss_pts_det_jabobian[i_gauss] * IntegrationPoints[i_gauss].Weight();
                         array_1d<double,3> interpolated_velocity = ZeroVector(3);
                         for (unsigned int n_node = 0; n_node < rGeom.PointsNumber(); n_node++){
-                            noalias( interpolated_velocity ) += N[n_node] * rGeom[n_node].FastGetSolutionStepValue(VELOCITY);
+                            noalias( interpolated_velocity ) += N[n_node] * (rGeom[n_node].FastGetSolutionStepValue(VELOCITY) -
+                             rGeom[n_node].FastGetSolutionStepValue(MESH_VELOCITY)) ;
                         }
                         inflow_over_boundary -= w_gauss * inner_prod( normal, interpolated_velocity );
                     }
 
                 // --- the condition is cut (2D)
-                } else if ( neg_count < rGeom.PointsNumber() && pos_count < rGeom.PointsNumber() ){
+                } else if (is_geometry_cut){
 
                     array_1d<double, 3> aux_velocity1, aux_velocity2;
 
                     // Generation of an auxiliary line that only covers the negative fraction ("transferring the splitting mechanism to 1D")
                     Line3D2<IndexedPoint>::Pointer p_aux_line = nullptr;
+                    for (unsigned int i = 0; i < rGeom.PointsNumber(); i++) {
+                        distance[i] = p_condition->GetGeometry()[i].FastGetSolutionStepValue( DISTANCE );
```

Comment:
```suggestion
                        distance[i] = rGeom[i].FastGetSolutionStepValue( DISTANCE );
```
You already have a reference.

Reply:
done